openbb_terminal/forecast/autoets_model.py

```diff
@@ -0,0 +1,132 @@
+# pylint: disable=too-many-arguments
+"""Automatic ETS (Error, Trend, and Seasonality) Model"""
+__docformat__ = "numpy"
+
+import logging
+from typing import Any, Union, Optional, List, Tuple
+
+import warnings
+import numpy as np
+import pandas as pd
+from statsforecast.models import ETS
+from statsforecast.core import StatsForecast
+
+from openbb_terminal.decorators import log_start_end
+from openbb_terminal.rich_config import console
+from openbb_terminal.forecast import helpers
+
+
+warnings.simplefilter("ignore")
+
+logger = logging.getLogger(__name__)
+
+
+@log_start_end(log=logger)
+def get_autoets_data(
+    data: Union[pd.Series, pd.DataFrame],
+    target_column: str = "close",
+    seasonal_periods: int = 7,
+    n_predict: int = 30,
+    start_window: float = 0.85,
+    forecast_horizon: int = 5,
+) -> Tuple[list[np.ndarray], List[np.ndarray], List[np.ndarray], Optional[float], Any]:
+
+    """Performs Automatic ETS forecasting
+    This is a wrapper around StatsForecast ETS;
+    we refer to this link for the original and more complete documentation of the parameters.
+
+
+        https://nixtla.github.io/statsforecast/models.html#ets
+
+    Parameters
+    ----------
+    data : Union[pd.Series, np.ndarray]
+        Input data.
+    target_column (str, optional):
+        Target column to forecast. Defaults to "close".
+    seasonal_periods: int
+        Number of seasonal periods in a year (7 for daily data)
+        If not set, inferred from frequency of the series.
+    n_predict: int
+        Number of days to forecast
+    start_window: float
+        Size of sliding window from start of timeseries and onwards
+    forecast_horizon: int
+        Number of days to forecast when backtesting and retraining historical
+
+    Returns
+    -------
+    list[float]
+        Adjusted Data series
+    list[float]
+        List of historical fcast values
+    list[float]
+        List of predicted fcast values
+    Optional[float]
+        precision
+    Any
+        Fit ETS model object.
+    """
+
+    use_scalers = False
+    # statsforecast preprocessing
+    # when including more time series
+    # the preprocessing is similar
+    ticker_series = data[["date", target_column]]
+    ticker_series.columns = ["ds", "y"]
+    ticker_series.insert(0, "unique_id", target_column)
+    ticker_series["ds"] = pd.to_datetime(ticker_series["ds"])
+
+    # Model Init
+    model_ets = ETS(
+        season_length=int(seasonal_periods),
+    )
+    fcst = StatsForecast(df=ticker_series, models=[model_ets], freq="B")
```

Comment:
Will this change in the future when we do not use "B" freq for time series? 

Does StatsForecast do an auto check if it doesn't happen to be "B"?

Reply:
Hey @martinb-bb! 

I think it would be better if there were an extra parameter in the models named `freq`, so that the user can include the frequency of their data. 

As far as I know, it is difficult for pandas to infer the frequency "B" because there are missing values in the data (weekends).

Should I include the `freq` parameter in this PR?

Reply:
Hey @FedericoGarza,  Just thinking aloud right now..

I would rather not add another kwarg for the time being, since we have so many. Some users are new to this field, and assuming they know frequency right out the gate can be overwhelming.

Instead, the goal is to make the menu robust under the hood. Our current code checks for `B` right off the bat, and if that fails, then tries to use pandas to assume `freq`. If that fails as well, we then resort to index of the Dataframe. 

found in helpers.py
```python 
try:
    ticker_series = TimeSeries.from_dataframe(**filler_kwargs)
except ValueError:
    # remove business days to allow base lib to assume freq
    filler_kwargs.pop("freq")
    ticker_series = TimeSeries.from_dataframe(**filler_kwargs)
except AttributeError:
    # Uses index if no date column
    console.print(
        "[red]Warning: no date found, assuming items evenly spaced.[/red]\n"
    )
    filler_kwargs.pop("time_col")
    ticker_series = TimeSeries.from_dataframe(**filler_kwargs)
```

Something like this would be ideal. If we include your library with the Darts wrapper, we can then do the same logic and it should work identically to the other models we have. Else, you can add some sort of custom logic in front of the `StatsForecast(df=ticker_series, models=[model_ets], freq="B")` before you pass in the time series? 

Only some thoughts...

Reply:
hey @martinb-bb! Thanks for sharing your thoughts. I agree with you, asking users for `freq` would be overwhelming.

I've changed the preprocessing step to use `helpers.get_series` before using `StatsForecast`,

https://github.com/OpenBB-finance/OpenBBTerminal/blob/ba75a1f80f9fb8d8c0ef18af0494b06d564a9d56/openbb_terminal/forecast/autoets_model.py#L75-L79

This ensures that the same logic is applied to the frequency between all models.🙌